rpc/src/module/test.rs

```diff
@@ -165,4 +159,69 @@ impl IntegrationTestRpc for IntegrationTestRpcImpl {
             Ok(hash.unpack())
         }
     }
+
+    fn generate_block_with_template(&self, block_template: BlockTemplate) -> Result<H256> {
+        let snapshot: &Snapshot = &self.shared.snapshot();
+        let consensus = snapshot.consensus();
+        let tip_header = snapshot.tip_header();
+        let mut seen_inputs = HashSet::new();
+
+        let txs: Vec<_> = block_template
+            .transactions
+            .iter()
+            .map(|tx_template| {
+                let transaction: packed::Transaction = tx_template.data.clone().into();
+                transaction.into_view()
+            })
+            .collect();
+
+        let transactions_provider = TransactionsProvider::new(txs.as_slice().iter());
+        let overlay_cell_provider = OverlayCellProvider::new(&transactions_provider, snapshot);
+
+        let rtxs: Vec<_> = txs.iter().filter_map(|tx| {
+            resolve_transaction(
+                tx.clone(),
+                &mut seen_inputs,
+                &overlay_cell_provider,
+                snapshot,
+            ).map_err(|err| {
+                error!(
+                    "resolve transactions error when generating block with block template, error: {:?}",
+                    err
+                );
+                err
+            }).ok()
+        }).collect();
+
+        let mut update_dao_template = block_template;
+        update_dao_template.dao = DaoCalculator::new(consensus, snapshot)
+            .dao_field(&rtxs, tip_header)
```

Comment:
assert `block_template.parent == tip_header.hash` before this line.

Reply:
I expect this rpc to be used to submit uncle blocks as well, so I don't want to add this condition。

Reply:
`fn dao_field`'s second parameter is "parent header". And here hard-code `tip_header` as parent header, which means the block's parent is `tip_header`.

If block_template.parent != tip_header.hash, the `fn dao_field` return the wrong value.

Reply:
aha, copy / paste error, now fixed, please review again, thanks.